pkg/cmd/dev/ui.go

```diff
@@ -248,3 +247,205 @@ Replaces 'make ui-lint'.`,
+			}
+
+			isStreamOutput, err := cmd.Flags().GetBool(streamOutputFlag)
+			if err != nil {
+				return err
+			}
+
+			if isWatch {
+				// Build prerequisites for watch-mode only. Non-watch tests are run through bazel, so it'll take care of this
+				// for us.
+				args := []string{
+					"build",
+					"//pkg/ui/workspaces/cluster-ui:cluster-ui",
+				}
+				logCommand("bazel", args...)
+				err = d.exec.CommandContextInheritingStdStreams(ctx, "bazel", args...)
+
+				if err != nil {
+					log.Fatalf("failed to build UI test prerequisites: %v", err)
+					return err
+				}
+
+				err = arrangeFilesForTestWatchers(d)
+				if err != nil {
+					// nolint:errwrap
+					return fmt.Errorf("unable to arrange files properly for watch-mode testing: %+v", err)
+				}
+
+				dirs, err := getUIDirs(d)
+				if err != nil {
+					log.Fatalf("unable to find cluster-ui and db-console directories: %v", err)
+					return err
+				}
+
+				nbExec := d.exec.AsNonBlocking()
+
+				argv0, argv := buildBazelYarnArgv(
+					"--silent",
+					"--cwd",
+					dirs.dbConsole,
+					"karma:watch",
+				)
+
+				env := append(os.Environ(), "BAZEL_TARGET=fake")
+				logCommand("yarn", args...)
+				err = nbExec.CommandContextWithEnv(ctx, env, argv0, argv...)
+				if err != nil {
+					// nolint:errwrap
+					return fmt.Errorf("unable to start db-console tests in watch mode: %+v", err)
+				}
+
+				argv0, argv = buildBazelYarnArgv(
+					"--silent",
+					"--cwd",
+					dirs.clusterUI,
+					"jest",
+					"--watch",
+				)
+				logCommand("yarn", args...)
+				env = append(os.Environ(), "BAZEL_TARGET=fake", "CI=1")
+				err = nbExec.CommandContextWithEnv(ctx, env, argv0, argv...)
+				if err != nil {
+					// nolint:errwrap
+					return fmt.Errorf("unable to start cluster-ui tests in watch mode: %+v", err)
+				}
+
+				// Wait for OS signals to cancel if we're not in test-mode
+				if !d.exec.IsDryrun() {
+					<-ctx.Done()
+				}
+			} else {
+				testOutputArg := d.getTestOutputArgs(
+					false, // stress
+					isVerbose,
+					false,                     // showLogs
+					isWatch || isStreamOutput, // streamOutput
+				)
+				args := append([]string{
+					"test",
+					"//pkg/ui/workspaces/db-console:karma",
+					"//pkg/ui/workspaces/cluster-ui:jest",
+				}, testOutputArg...)
+
+				if isDebug {
+					args = append(args, "--subcommands")
+				}
+
+				logCommand("bazel", args...)
+				err = d.exec.CommandContextInheritingStdStreams(ctx, "bazel", args...)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+
+	testCmd.Flags().Bool(watchFlag, false, "watch source and test files for changes and rerun tests")
+	testCmd.Flags().BoolP(verboseFlag, "v", false, "show all test results when tests complete")
+	testCmd.Flags().Bool(streamOutputFlag, false, "stream test output during run (default: true with --watch)")
+
+	return testCmd
+}
+
+func buildBazelYarnArgv(argv ...string) (string, []string) {
+	return "bazel", append([]string{
```

Comment:
I don't think returning `"bazel"` as the first argument here makes a great deal of sense. Most places we just hard-code it.

Reply:
That's fair!  It felt nice to not have this function handle everything, but I can pull that out.